packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -541,9 +540,8 @@ InternalModel.prototype = {
   },
 
   setId: function(id) {
+    Ember.assert('Object is a newly created model instance', this.isNew);
```

Comment:
`isNew` is a function. `Ember.assert('Object is a newly created model instance', this.isNew());`